kolibri/core/assets/src/core-app/constructorExport.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * This exports the API for the core Kolibri app.
+ */
+
+const apiSpec = require('./apiSpec').apiSpec;
+const keys = require('./apiSpec').keys;
+
+
+const constructorExport = () => {
+  /*
+   * Function for building the object that populates the kolibri global object API.
+   */
+  const exportObj = {};
+  const recurseObjectKeysAndImport = (obj, path = []) => {
+    Object.keys(obj).forEach((key) => {
+      if (keys.indexOf(key) === -1) {
+        recurseObjectKeysAndImport(obj[key], path.concat(key));
+      }
+    });
+    // Don't allow privileged keys in the top namespace, as, logically, that would overwrite
```

Comment:
Oh, that should mean 'requireName' or 'module' - should probably find a better word.